### PR TITLE
Set `CREATE_NO_WINDOW` flag when executing Git hooks

### PR DESCRIPTION
**Pull request by icorbrey**
This Pull Request makes changes to accommodate GUI functionality in GitButler (see: https://github.com/gitbutlerapp/gitbutler/issues/4982).

It changes the following:
- Set the `CREATE_NO_WINDOW` flag when executing Git hooks on Windows.

I followed the checklist:
- [ ] I added unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [x] I added an appropriate item to the changelog

### CODE REVIEW

**Comment by icorbrey** [comment]
I've tested this functionality using a dev build of GitButler with the `git2-hooks` dependency pointed at my local copy of `gitui`. Please let me know if I need to do anything else to get this merged, it bothers me when using GitButler with Lefthook and I'd like to be able to get it fixed!

**Comment by extrawurst** [comment]
Thanks for the contribution! Can you fix the ci issues?

**Comment by icorbrey** [comment]
Sure thing!

**Comment by icorbrey** [comment]
@extrawurst that should fix it, I had forgotten to ensure this was a noop when you're not on Windows

**Comment by icorbrey** [comment]
@extrawurst out of curiosity, do you have an idea of when the next release will be? Also, I believe the CI workflow requires your approval again.

File: git2-hooks/src/hookspath.rs [hunk 1]
```diff
@@ -197,3 +201,24 @@ fn find_bash_executable() -> Option<PathBuf> {
...
+
+trait CommandExt {
+	fn with_no_window(&mut self) -> &mut Self;
+}
+
+impl CommandExt for Command {
+	/// On Windows, CLI applications that aren't the window's subsystem will
+	/// create and show a console window that pops up next to the main
+	/// application window when run. We disable this behavior by setting the
+	/// `CREATE_NO_WINDOW` flag.
```
Comment (extrawurst) [inline git2-hooks/src/hookspath.rs:213]: can we add a link here to some place that documents this magic number? maybe even use a local constant that is named equally `CREATE_NO_WINDOW`?

**Comment by extrawurst** [comment]
@icorbrey since this is a separate crate i can cut a release as soon as this is merged

File: git2-hooks/src/hookspath.rs (hunk 1 above)
Comment (icorbrey) [reply git2-hooks/src/hookspath.rs:223]: I went ahead and added the following to `CommandExt`:

```rs
/// The process is a console application that is being run without a
/// console window. Therefore, the console handle for the application is
/// not set.
///
/// This flag is ignored if the application is not a console application,
/// or if it used with either `CREATE_NEW_CONSOLE` or `DETACHED_PROCESS`.
///
/// See: https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags
const CREATE_NO_WINDOW: u32 = 0x0800_0000;
```

This is just the documentation for `CREATE_NO_WINDOW` from the linked Microsoft docs. Are you happy with this?

**Comment by extrawurst** [comment]
released git2-hooks `v0.4.0`

**Pull request merged**